### PR TITLE
Distance map

### PR DESCRIPTION
**Pull request by frheault**
# Quick description

Simple script to compute the minimum distance between two masks. This could be useful to compute a lesion to bundle distance to create a NIFTI image and then color a bundle.
(The white sphere is a <lesion>, rendered in 3D in MI-Brain, it was initially a binary mask.

image

...

## Type of change

Check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Provide data, screenshots, command line to test (if relevant)

...

# Checklist

- [x] My code follows the style guidelines of this project (run [autopep8](https://pypi.org/project/autopep8/))
- [ ] I added relevant citations to scripts, modules and functions docstrings and descriptions
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I moved all functions from the script file (except the argparser and main) to scilpy modules
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: scripts/scil_volume_distance_map.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+is the Euclidean distance from each voxel of the first mask to the closest
+voxel of the second mask.
+It's worst case scenario is two very big masks that are far appart or a very
+small mask that is completely inside a very big mask.
+
+Take this command as an example:
+  scil_volume_distance_map.py AF_L.nii.gz brain_mask.nii.gz \
+    AF_L_to_brain_mask.nii.gz
+
+We have a bundle and brain mask, the first is 100% inside the second.
```
Comment (AntoineTheb) [inline scripts/scil_volume_distance_map.py:15]: The example should feature "positive" examples instead of edge cases.

File: scripts/scil_volume_distance_map.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Compute distance map between two binary masks using KDTree. The distance map
+is the Euclidean distance from each voxel of the first mask to the closest
+voxel of the second mask.
+It's worst case scenario is two very big masks that are far appart or a very
```
Comment (AntoineTheb) [inline scripts/scil_volume_distance_map.py:8]: "Corner cases include two very big masks that are far apart or one mask inside another."

File: scripts/scil_volume_distance_map.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Compute distance map between two binary masks using KDTree. The distance map
```
Comment (AntoineTheb) [inline scripts/scil_volume_distance_map.py:5]: Not sure the KDTree part is relevant, for a medically-trained user for example.

File: scripts/scil_volume_distance_map.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    img_1 = nib.load(args.in_mask_1)
+    img_2 = nib.load(args.in_mask_2)
+
+    mask_1 = get_data_as_mask(img_1)
+    mask_2 = get_data_as_mask(img_2)
+    logging.debug(f'Loaded two masks with {np.count_nonzero(mask_1)} and '
+                  f'{np.count_nonzero(mask_2)} voxels')
+
+    # Compute distance map using KDTree
```
[the remark this answers is not in the document]
Comment (AntoineTheb) [reply scripts/scil_volume_distance_map.py:74]: This way, unit tests could be added to handle the corner cases described in the docstring.

**Review by AntoineTheb — CHANGES_REQUESTED** [review]

File: scripts/scil_volume_distance_map.py (hunk 1 above)
Comment (frheault) [reply scripts/scil_volume_distance_map.py:15]: I swap the order to start with the normal example, but I left the edge case since it is an important concept for distance map.

File: scripts/scil_volume_distance_map.py (hunk 2 above)
Comment (frheault) [reply scripts/scil_volume_distance_map.py:8]: They are not corner cases, they are slowest case scenario, I changed the phrase accordingly

**Comment by frheault** [comment]
@arnaudbore I would merge this tomorrow and anyway Zineb and I are doing another PR immediately that will use this new code. So we will change a bit the code if required.
(It is working great)
